### PR TITLE
Switch podman image push handlers to use abi

### PR DESCRIPTION
**Pull request by rhatdan**
Change API Handlers to use the same functions that the
local podman uses.

At the same time:

Cleanup and pass proper bindings.  Remove cli options from
podman-remote push.  Cleanup manifest push.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Please use the `X-Registry-Auth` header to pass credentials as other endpoints do.  Passing the credentials via the parameters may be a security issue. There are convenience functions in `pkg/auth` to easily add and extract credentials on the client and server.

File: pkg/api/handlers/libpod/manifests.go [hunk 1]
```diff
@@ -148,7 +148,9 @@ func ManifestPush(w http.ResponseWriter, r *http.Request) {
 		All         bool   `schema:"all"`
 		Destination string `schema:"destination"`
 		Format      string `schema:"format"`
+		Password    string `schema:"password"`
 		TLSVerify   bool   `schema:"tlsVerify"`
+		Username    string `schema:"username"`
```
Comment (vrothberg) [inline pkg/api/handlers/libpod/manifests.go:153]: Still some left.

Comment (vrothberg) [reply pkg/api/handlers/libpod/manifests.go:153]: I think the changes in this file can be reverted.

File: pkg/api/server/register_manifest.go [hunk 2]
```diff
@@ -151,6 +151,14 @@ func (s *APIServer) registerManifestHandlers(r *mux.Router) error {
...
 	//    description: push all images
 	//    type: boolean
+	//  - in: query
+	//    name: username
+	//    type: string
+	//    description: username for authenticating against the registry.
+	//  - in: query
+	//    name: password
+	//    type: string
+	//    description: password for authenticating against the registry.
```
Comment (vrothberg) [inline pkg/api/server/register_manifest.go:161]: And these.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@vrothberg @mheon PTAL
This is passing tests, and should be ready to merge.

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks @rhatdan !

File: docs/source/markdown/podman-push.1.md [hunk 3]
```diff
@@ -91,19 +91,19 @@ solely for scripting compatibility.
 #### **--format**, **-f**=*format*
 
 Manifest Type (oci, v2s1, or v2s2) to use when pushing an image to a directory using the 'dir:' transport (default is manifest type of source)
-Note: This flag can only be set when using the **dir** transport
+Note: This flag can only be set when using the **dir** transport. (Not available for remote commands)
```
Comment (mheon) [inline docs/source/markdown/podman-push.1.md:94]: Suggest rewording all of these to "Not available with the remote client"

Comment (rhatdan) [reply docs/source/markdown/podman-push.1.md:94]: Mheon I am going to merge this and then change the man pages as a docs only change.
